pkg/controllers/clusterresourceplacement/controller.go

```diff
@@ -678,7 +739,7 @@ func parseResourceGroupHashFromAnnotation(s *fleetv1beta1.ClusterResourceSnapsho
 }
 
 // TODO: handle multiple resourceSnapshots
-func (r *Reconciler) buildPlacementStatus(crp *fleetv1beta1.ClusterResourcePlacement, latestSchedulingPolicySnapshot *fleetv1beta1.ClusterSchedulingPolicySnapshot, latestResourceSnapshot *fleetv1beta1.ClusterResourceSnapshot) (*fleetv1beta1.ClusterResourcePlacementStatus, error) {
+func (r *Reconciler) buildPlacementStatus(ctx context.Context, crp *fleetv1beta1.ClusterResourcePlacement, latestSchedulingPolicySnapshot *fleetv1beta1.ClusterSchedulingPolicySnapshot, latestResourceSnapshot *fleetv1beta1.ClusterResourceSnapshot) (*fleetv1beta1.ClusterResourcePlacementStatus, error) {
 	status := &fleetv1beta1.ClusterResourcePlacementStatus{
```

Comment:
(The thread is about the section below, new line 750-764)

Reply:
The handleUpdate method already performs selectResources routine (and it has to do it every time); here it just deserializes what it just serializes, and this is expensive.

Reply:
Also, a nit about the `buildScheduledCondition` method: for the unknown type, it might be better for the message to be a little bit more user-facing, i.e., "scheduling is in progress", rather than "record the reason ..."

Reply:
good catch! forget that we've already handled selectResources.